server/events.go

```diff
@@ -331,16 +334,44 @@ RESET:
 			c.mu.Lock()
 
 			// Prep internal structures needed to send message.
-			c.pa.subject = []byte(pm.sub)
-			c.pa.size = len(b)
-			c.pa.szb = []byte(strconv.FormatInt(int64(len(b)), 10))
-			c.pa.reply = []byte(pm.rply)
+			c.pa.subject, c.pa.reply = []byte(pm.sub), []byte(pm.rply)
+			c.pa.size, c.pa.szb = len(b), []byte(strconv.FormatInt(int64(len(b)), 10))
```

Comment:
No big deal, but looking down, it seems that if there is compression, c.setHeader() will set some of the c.pa fields based on the new header content, etc.. So this line here will be undone if there is compression and header added. Not asking to change, but making sure I understand this.

Reply:
Yes that is correct. It needs a properly formed message with /r/n and any existing header info in the c.pa before we call into setHeader, hence the ordering.